gno/p/role_manager/role_manager.gno

```diff
@@ -0,0 +1,406 @@
+package role_manager
+
+// TODO: RENAME FUNCTIONS (HAS PERMISSIONS -> UserHasPermission, HAS ROLE -> UserHasRole, etc)
+// TODO: CREATE HAS PERMISSION and HAS ROLE functions that just look if role or permission exists not if user has it
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ownable"
+)
+
+// well known roles
+const (
+	ROLE_USER = "user"
+)
+
+// well known permissions
+const (
+	PERM_ALL         = "all"
```

Comment:
for this one, I would suggest you to support a "glob" syntax instead; this way you can have things like `role:*`, or any other `area:*` to grant all permissions in a specific area.